### PR TITLE
Rudderstack cleanup 2

### PR DESCRIPTION
**Pull request by rsdy**
This makes the user `tracking_id` consistent and enforced by Rust APIs. A number of arbitrary `user_id` fields have also been centralized around a single behaviour.

### CODE REVIEW

File: server/bleep/src/analytics.rs [hunk 1]
```diff
@@ -44,11 +46,18 @@ pub struct PackageMetadata {
...
     options: Option<HubOptions>,
+
+    /// Rudderstack client
     client: RudderAnalytics,
+
+    /// User-specific store
+    user_store: PersistedState<scc::HashMap<String, UserState>>,
+
+    /// Application-wide tracking seed
+    tracking_seed: PersistedState<ApplicationSeed>,
```
Comment (ggordonhall) [inline server/bleep/src/analytics.rs:60]: Can you update this comment to describe what the `tracking_seed` is?

File: server/bleep/src/analytics.rs [hunk 2]
```diff
@@ -57,52 +66,82 @@ pub struct HubOptions {
...
+        state: &StateSource,
+        tracking_seed: impl Into<Option<String>>,
+        key: String,
+        data_plane: String,
+    ) -> anyhow::Result<Arc<Self>> {
+        Self::new_with_options(state, tracking_seed, key, data_plane, None)
     }
 
-    pub fn new_with_options(key: String, data_plane: String, options: HubOptions) -> Arc<Self> {
+    pub fn new_with_options(
```
Comment (ggordonhall) [inline server/bleep/src/analytics.rs:89]: Do we need both these `new` methods?

File: server/bleep/src/analytics.rs [hunk 3]
```diff
@@ -57,52 +66,82 @@ pub struct HubOptions {
...
-        let hub = Self {
+        Ok(Self {
             client,
-            options: Some(options),
-        };
-        let _ = HUB.set(Arc::new(hub));
-        RudderHub::get().unwrap()
+            options: options.into(),
+            user_store: state.load_or_default("user_tracking")?,
+            tracking_seed: state.load_state_or("application_seed", tracking_seed.into())?,
```
Comment (ggordonhall) [inline server/bleep/src/analytics.rs:101]: Bit unclear on the semantics here. What's the difference between:
- `tracking_seed`
- `application_seed`
- `tracking_id`

**Review by ggordonhall — APPROVED** [review]

File: server/bleep/src/analytics.rs [hunk 4]
```diff
@@ -44,65 +56,121 @@ pub struct PackageMetadata {
...
+                let id = self
+                    .user_store
+                    .entry(username.clone())
+                    .or_default()
+                    .get()
+                    .tracking_id();
+                _ = self.user_store.store();
+                id
+            }
+            None => self.device_id.to_string(),
```
[the remark this answers is not in the document]
Comment (rsdy) [reply server/bleep/src/analytics.rs:123]: When a user with a local installation chooses not to authenticate to GitHub. This configuration is currently not exposed through the frontend, but the state can happen on the backend.

**Pull request merged**